docs/architecture-decision-records/0020-aggregation-for-tdr-preingest.md

```diff
@@ -0,0 +1,40 @@
+# 20. Aggregation for TDR Preingest
+
+**Date:** 2024-08-14
+
+## Context
+
+The first source to ingest into DR2 was parsed court documents transferred via TDR and parsed by TRE. The upstream workflow for this type of record enforces a 1-1 relationship between a transfer consignment and a record. We're now ready to create a preingest route for standard TDR transfers.
+
+Standard TDR transfers contain multiple files within a consignment; the count and size of the files is at the discretion of the transferring user. While the Preservation System ingest process is most efficient when ingesting batches, as opposed to individual files, the size of the batches are unlikely to match the user decisions upstream of DR2; we want to design a preingest process for TDR to be most efficient for the Preservation System.
+
+## Decision
+
+We will implement an aggregation process as part of our TDR Preingest workflow, we will request TDR notify us of new records for ingest with a message per record to allow us to aggregate. Our aggregation process will use the same AWS services as our ingest process - SQS, Lambda, Step Functions, and DynamoDB.

```

Comment:
2. Changed to
> We will implement an aggregation process as part of our TDR Preingest workflow; we will request TDR notify us of new records for ingest with a message per record to allow us to aggregate records into ingest batches of a configurable size.

3. I could but the schema is still undecided.

Reply:
But what is it about the `message` specifically that will enable us to "aggregate records into ingest batches"?

Reply:
Added
> This is opposed to a message per transfer consignment, which could contain multiple records.

Reply:
Ahh, now that I think about it, it's just clicked, you meant an SNS/SQS message per record. Because I've been working on the `messageId`/`correlationId` task, I'm thinking of the `message` attribute that goes into the lock table. Sorry 🤦 ; I blame the fact that it is Friday afternoon 😄.